datalake/backend/tests/test_processing_stack.py

```diff
@@ -0,0 +1,133 @@
+import json
+import logging
+import os
+import time
+
+from pytest import mark
+
+from .utils import S3Object, any_safe_file_path
+
+ENV = os.environ.get("DEPLOY_ENV", "dev")
+
+APPLICATION_TAG_KEY = "ApplicationName"
+APPLICATION_TAG_VAL = "geospatial-data-lake"
+
+ENVIRONMENT_TAG_KEY = "EnvironmentType"
+ENVIRONMENT_TAG_VAL = ENV
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class NoStateMachineFound(Exception):
+    pass
+
+
+class NoComputeEnvironmentFound(Exception):
+    pass
+
+
+def get_compute_environment(batch_client):
+    compute_environment_detection_response = batch_client.describe_compute_environments()
+
+    # make sure there are no more records in next pages
+    assert compute_environment_detection_response.get("nextToken", None) is None
+
+    for compute_environment in compute_environment_detection_response["computeEnvironments"]:
+        if (
+            compute_environment["tags"][ENVIRONMENT_TAG_KEY] == ENVIRONMENT_TAG_VAL
+            and compute_environment["tags"].get(APPLICATION_TAG_KEY, None) == APPLICATION_TAG_VAL
+        ):
+            datalake_compute_environment = compute_environment
+            logger.info("Datalake Compute Environment: %s", datalake_compute_environment)
+
+            return datalake_compute_environment
+
+    raise NoStateMachineFound(APPLICATION_TAG_VAL, ENVIRONMENT_TAG_VAL)
+
+
+def launch_compute_environment_resources(batch_client, datalake_compute_environment):
```

Comment:
How big is the difference between running this first and not running it? Comparing runs https://github.com/linz/geospatial-data-lake/runs/1718745340 and https://github.com/linz/geospatial-data-lake/runs/1718808531 it looks like a run without this takes 2m10s.

Reply:
You mean without running `launch_compute_environment_resources` function ? It can take 15 minutes  to launch compute resources without setting min and desired CPU to non zero value. Based on my experience it is very significant diference